src/runtime/security/kinit_context.cpp

```diff
@@ -74,14 +74,15 @@ error_s check_configuration()
 class kinit_context : public utils::singleton<kinit_context>
 {
 public:
-    kinit_context() : _opt(nullptr) {}
-    virtual ~kinit_context();
+    ~kinit_context();
 
     // implementation of 'kinit -k -t <keytab_file> <principal>'
     error_s kinit();
-    const std::string username() const { return _user_name; }
+    const std::string &username() const { return _user_name; }
 
 private:
+    kinit_context() : _opt(nullptr) {}
```

Comment:
`_opt` is initialized as `nullptr ` in line 108, no need to re-initialize it here.